eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Pool.hs

```diff
@@ -94,14 +96,11 @@ data ShelleyPoolPredFailure era
       !(KeyHash 'StakePool (EraCrypto era)) -- KeyHash which cannot be retired since it is not registered
   | StakePoolRetirementWrongEpochPOOL
       !EpochNo -- Current Epoch
```

Comment:
I think using the mismatch for the lower bound could be useful as well:
```suggestion
      !(Mismatch 'RelGT EpochNo)
```

Otherwise it is possible for the second `Mismatch` to not be a "mismatch", since it was the first condition that was at fault here.